crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -0,0 +1,220 @@
+use rustpython_parser::ast::{Arguments, Constant, Expr, ExprKind, Operator, Unaryop};
+
+use ruff_macros::{define_violation, derive_message_formats};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+define_violation!(
+    pub struct TypedArgumentSimpleDefaults;
+);
+/// PYI011
+impl Violation for TypedArgumentSimpleDefaults {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Only simple default values allowed for typed arguments")
+    }
+}
+
+define_violation!(
+    pub struct ArgumentSimpleDefaults;
+);
+/// PYI014
+impl Violation for ArgumentSimpleDefaults {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Only simple default values allowed for arguments")
+    }
+}
+
+const ALLOWED_ATTRIBUTES_IN_DEFAULTS: &[&[&str]] = &[
+    &["sys", "stdin"],
+    &["sys", "stdout"],
+    &["sys", "stderr"],
+    &["sys", "version"],
+    &["sys", "version_info"],
+    &["sys", "platform"],
+    &["sys", "executable"],
+    &["sys", "prefix"],
+    &["sys", "exec_prefix"],
+    &["sys", "base_prefix"],
+    &["sys", "byteorder"],
+    &["sys", "maxsize"],
+    &["sys", "hexversion"],
+    &["sys", "winver"],
+];
+
+fn is_valid_default_value_with_annotation(default: &Expr, checker: &Checker) -> bool {
+    match &default.node {
+        ExprKind::Constant {
+            value: Constant::Ellipsis | Constant::None,
+            ..
+        } => {
+            return true;
+        }
+        ExprKind::Constant {
+            value: Constant::Str(..),
+            ..
+        } => return checker.locator.slice(&Range::from_located(default)).len() <= 50,
+        ExprKind::Constant {
+            value: Constant::Bytes(..),
+            ..
+        } => return checker.locator.slice(&Range::from_located(default)).len() <= 50,
+        ExprKind::Constant {
+            value: Constant::Int(..),
+            ..
+        } => {
+            return checker.locator.slice(&Range::from_located(default)).len() <= 10;
```

Comment:
I changed these to look at the size of the actual int in source code, which I _think_ mirrors the intent of `flake8-pyi`.